browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -100,23 +100,28 @@ void IPFSTabHelper::IPFSResourceLinkResolved(const GURL& ipfs) {
   UpdateLocationBar();
 }
 
-void IPFSTabHelper::DNSLinkResolved(const GURL& ipfs) {
+void IPFSTabHelper::DNSLinkResolved(const GURL& ipfs, bool is_gateway_url) {
   DCHECK(!ipfs.is_valid() || ipfs.SchemeIs(kIPNSScheme));
   ipfs_resolved_url_ = ipfs.is_valid() ? ipfs : GURL();
-  if (ipfs.is_valid() && pref_service_->GetBoolean(kIPFSAutoRedirectDNSLink)) {
+  bool should_redirect =
+      (is_gateway_url && pref_service_->GetBoolean(kIPFSAutoRedirectGateway)) ||
+      (!is_gateway_url && pref_service_->GetBoolean(kIPFSAutoRedirectDNSLink));
+  if (ipfs.is_valid() && should_redirect) {
     LoadUrl(GetIPFSResolvedURL());
     return;
   }
   UpdateLocationBar();
 }
 
 void IPFSTabHelper::HostResolvedCallback(
+    const GURL& current,
+    const GURL& check_url,
+    bool is_gateway_url,
     absl::optional<std::string> x_ipfs_path_header,
     const std::string& host,
     const absl::optional<std::string>& dnslink) {
-  GURL current = web_contents()->GetURL();
-
-  if (current.host() != host || !current.SchemeIsHTTPOrHTTPS())
+  if (current.host() != web_contents()->GetURL().host() ||
```

Comment:
GetURL() is deprecated.
As long as code around is being changed can we pick right substitute?